src/testing/flow_solver.h

```diff
@@ -36,10 +43,10 @@ class FlowSolver : public TestsBase
     const std::string unsteady_data_table_filename_with_extension; ///< Filename (with extension) for the unsteady data table
         
     /// Displays the flow setup parameters
-    virtual void display_flow_solver_setup(const Parameters::AllParameters *const all_param) const;
+    virtual void display_flow_solver_setup() const;
 
     /// Pure virtual function to generate the grid
-    virtual void generate_grid(std::shared_ptr<dealii::parallel::distributed::Triangulation<dim>> grid) const = 0;
+    virtual void generate_grid(std::shared_ptr<Triangulation> grid) const = 0;
```

Comment:
I'd guess that almost everything except run_test can be made protected

Reply:
Agreed

Reply:
The constructors and destructors also needed to be public. This is fixed in the most recent commit.